protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
I don't agree with this.

First, why only the TB case?  What about if you try to register a different key from one that's already made its way into a signed root?  That would return a `ReqNameExisted` and a differing key, and similarly fail to verify the auth path.

And, rightly so.  It might be nice if the returned error was `ErrBindingsDiffer` or something, but this is still a case we want to fail a check on, no?